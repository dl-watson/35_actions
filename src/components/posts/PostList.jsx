import React from "react";
import { useSelector } from "react-redux";
import { getPosts } from "../../selectors/postSelectors";
import Post from "./Post";

const PostList = () => {
  const { posts } = useSelector(getPosts);

  const list = posts.map((post, i) => {
    return (
      <li key={post.id}>
        <Post {...post} postIndex={i} />
      </li>
    );
  });
  return <ul data-testid="post-container">{list}</ul>;
};

export default PostList;
