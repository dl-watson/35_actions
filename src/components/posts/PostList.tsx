import React from "react";
import { useSelector } from "react-redux";
import { getPosts } from "../../selectors/postSelectors";
import Post from "./Post";

type PostProps = {
  id: number, 
  title: string, 
  body: string,
}

const PostList = () => {
  const { posts } = useSelector(getPosts);

  const list = posts.map((post: PostProps, i: number) => {
    return (
      <li key={post.id}>
        <Post {...post} postIndex={i} />
      </li>
    );
  });
  return <ul data-testid="post-container">{list}</ul>;
};

export default PostList;
