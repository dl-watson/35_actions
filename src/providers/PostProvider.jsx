import React, { createContext, useReducer } from "react";
import { initialState, postReducer } from "../reducers/postReducer";

const PostContext = createContext(null);

export const PostProvider = ({ children }) => {
  const [state, dispatch] = useReducer(postReducer, initialState);

  return <PostContext.Provider>{children}</PostContext.Provider>;
};
