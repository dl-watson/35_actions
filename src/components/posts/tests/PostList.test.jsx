import React from "react";
import { render, cleanup } from "@testing-library/react";
import PostList from "../PostList";
import { PostProvider } from "../../../providers/PostProvider";

describe("PostList component", () => {
  afterEach(() => cleanup());
  it("renders PostList", () => {
    const { asFragment } = render(
      <PostProvider>
        <PostList />
      </PostProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
