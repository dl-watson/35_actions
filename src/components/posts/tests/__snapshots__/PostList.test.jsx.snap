// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component renders PostList 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
